@here/harp-mapview/lib/geometry/overlayOnElevation.ts

```diff
@@ -20,6 +29,58 @@ function overlayObject(object: TileObject, displacementMap: THREE.DataTexture):
     }
 }
 
+const worldCoords = new THREE.Vector3();
+
+/**
+ * Overlays the specified coordinates on top of elevation data if available.
+ *
+ * @param tileCoords World coordinates relative to the `tile` center.
+ * @param elevationProvider Used to get elevation data.
+ * @param tile The tile to which the coordinates are relative.
+ */
+function overlayPosition(
+    tileCoords: THREE.Vector3,
+    elevationProvider: ElevationProvider,
+    tile: Tile
+) {
+    worldCoords.copy(tileCoords).add(tile.center);
+    const geoCoords = tile.mapView.projection.unprojectPoint(worldCoords);
+    const height = elevationProvider.getHeight(geoCoords);
+
+    if (height !== undefined) {
+        geoCoords.altitude = height;
+        tile.mapView.projection.projectPoint(geoCoords, worldCoords);
+        tileCoords.copy(worldCoords).sub(tile.center);
+    }
+}
+
+/**
+ * Overlays a text element on top of elevation data if available.
+ *
+ * @param textElement The text element whose geometry will be overlaid.
+ * @param elevationProvider Used to get elevation data.
+ * @param tile The tile the text element belongs to.
+ */
+function overlayTextElement(
```

Comment:
I guess it would be better if the labels fade in _after_ they have been put on the overlay, to reduce the issue of labels moving.